app/models/mixins/virtual_total_mixin.rb

```diff
@@ -16,7 +16,7 @@ module ClassMethods
     #
     def virtual_total(name, relation, options = {})
       define_method(name) do
-        send(relation).try(:size) || 0
+        attributes[name.to_s] || send(relation).try(:size) || 0
```

Comment:
In the past you typically wanted to avoid `attributes[]` because it `dup`ed the entire attributes list.  I think there is a different way to get at the attribute that is more efficient (perhaps just `.send`).